pandas/core/reshape/merge.py

```diff
@@ -1385,6 +1386,15 @@ def _maybe_coerce_merge_keys(self) -> None:
                 if lk.dtype.kind == rk.dtype.kind:
                     continue
 
+                if is_extension_array_dtype(lk.dtype) and not is_extension_array_dtype(
+                    rk.dtype
+                ):
+                    ct = find_common_type([lk.dtype, rk.dtype])
```

Comment:
Do we know that this will always return an `ExtensionDtype`? Might be good to first check that it is one

Reply:
Yeah one more check won't hurt. Updated